web/src/engine/websites/Anchira.ts

```diff
@@ -0,0 +1,102 @@
+import { Tags } from '../Tags';
+import icon from './Anchira.webp';
+import { type Chapter, DecoratableMangaScraper, Manga, Page, type MangaPlugin } from '../providers/MangaPlugin';
+import * as Common from './decorators/Common';
+import { FetchJSON } from '../platform/FetchProvider';
+
+type APIMangas = {
+    entries: APIManga[]
+}
+
+type APIManga = {
+    id: number,
+    key: string,
+    title: string
+    hash?: string,
+    names?: string[]
+}
+
+@Common.ChaptersUniqueFromManga()
+@Common.ImageAjax()
+export default class extends DecoratableMangaScraper {
+
+    private readonly apiUrl = `${this.URI}api/v1/`;
+    private readonly imageCDN = 'https://kisakisexo.xyz';
+
+    public constructor() {
+        super('anchira', `Anchira`, 'https://anchira.to', Tags.Language.English, Tags.Media.Manga, Tags.Source.Aggregator, Tags.Rating.Pornographic);
+    }
+
+    public override get Icon() {
+        return icon;
+    }
+
+    public override ValidateMangaURL(url: string): boolean {
+        return new RegExp(`^${this.URI.origin}/g/[\\d]+/[^/]+$`).test(url);
+    }
+
+    public override async FetchManga(provider: MangaPlugin, url: string): Promise<Manga> {
+        const matches = url.match(/\/g\/(\d+)\/([^/]+)/);
+        const id = matches[1];
+        const key = matches[2];
```

Comment:
May just get id + key of the url path in a single capture group, no need for two separate capture groups